xenserver/tests/test_tasks.py

```diff
@@ -2,74 +2,176 @@
-        self.local_sr = xenserver.add_SR('Local storage', 'lvm')
-        isos = xenserver.add_SR('ISOs', 'iso')
+@pytest.mark.django_db
+class TestCreateVM(object):
+    def setup_xs_storage(self, xenserver, iso_names=("installer.iso",)):
+        self.local_SR = xenserver.add_SR('Local storage', 'lvm')
+        self.iso_SR = xenserver.add_SR('ISOs', 'iso')
         for iso_name in iso_names:
-            xenserver.add_VDI(isos, iso_name)
+            xenserver.add_VDI(self.iso_SR, iso_name)
+
+    def setup_xs_networks(self, xenserver):
+        self.pub_net = xenserver.add_network(bridge="xenbr0")
+        self.prv_net = xenserver.add_network(bridge="xenbr1")
+        self.pub_PIF = xenserver.add_PIF(
+            self.pub_net, device="eth0", gateway="10.1.2.3")
+        self.prv_PIF = xenserver.add_PIF(self.prv_net, device="eth1")
+
+    def setup_template(self, name, cores=1, memory=2048, diskspace=10240,
+                       iso="installer.iso"):
+        return Template.objects.create(
+            name=name, cores=cores, memory=memory, diskspace=diskspace,
+            iso=iso)
+
+    def setup_vm(self, name, template, status="Provisioning", **kw):
+        params = {
+            "sockets": template.cores,
+            "memory": template.memory,
+        }
+        params.update(kw)
+        return XenVM.objects.create(
+            name="foovm", status="Provisioning", template=template, **params)
+
+    def extract_VIFs(self, xenserver, VM_ref, spec):
+        """
+        Get the VIFs for the given VM and match them to a list of (network,
+        VIF) ref pairs.
+        """
+        assert MatchSorted(spec) == xenserver.list_network_VIFs_for_VM(VM_ref)
+
+    def extract_VBDs(self, xenserver, VM_ref, spec):
+        """
+        Get the VBDs for the given VM and match them to a list of (SR, VBD)
+        ref pairs.
+        """
+        assert MatchSorted(spec) == xenserver.list_SR_VBDs_for_VM(VM_ref)
+
+    def expected_vm(self, template, VIFs, VBDs, **kw):
+        """
+        Build an ExpectedXenServerVM object with some default parameters.
+        """
+        params = {
+            "PV_args": " -- quiet console=hvc0",
+            "name_label": "None.None",
+            "VCPUs_max": "1",
+            "VCPUs_at_startup": "1",
+            "memory_static_max": str(template.memory*1024*1024),
+            "memory_dynamic_max": str(template.memory*1024*1024),
+            "suspend_SR": self.local_SR,
+        }
+        return ExpectedXenServerVM(
+            VIFs=MatchSorted(VIFs), VBDs=MatchSorted(VBDs), **params)
 
     def test_create_vm_simple(self):
         """
         We can create a new VM using mostly default values.
+        """
+        xenserver = FakeXenServer()
+        self.setup_xs_storage(xenserver)
+        self.setup_xs_networks(xenserver)
+        session = xenserver.getSession()
+        template = self.setup_template("footempl")
+        vm = self.setup_vm("foovm", template)
```

Comment:
Some tests will want it to be different. I just don't have any of those tests yet.